pkg/provisioning/validate.go

```diff
@@ -0,0 +1,83 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provisioning
+
+import (
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/multierror"
+)
+
+var (
+	ErrMandatoryField = cerrors.New("mandatory field not specified")
+	ErrInvalidField   = cerrors.New("invalid field value")
+)
+
+const (
+	StatusRunning   = "running"
+	StatusStopped   = "stopped"
+	TypeConnector   = "source"
+	TypeDestination = "destination"
+)
+
+// ValidatePipelinesConfig validates config field values for a pipeline
+func ValidatePipelinesConfig(cfg PipelineConfig) error {
+	var err, tmpErr error
+	if cfg.Status != StatusRunning && cfg.Status != StatusStopped {
+		err = multierror.Append(err, cerrors.Errorf("\"status\" is invalid: %w", ErrInvalidField))
+	}
+	tmpErr = validateConnectorsConfig(cfg.Connectors)
+	if tmpErr != nil {
+		err = multierror.Append(err, tmpErr)
+	}
+	tmpErr = validateProcessorsConfig(cfg.Processors)
+	if tmpErr != nil {
+		err = multierror.Append(err, tmpErr)
+	}
+	return err
+}
+
+// validateConnectorsConfig validates config field values for connectors
+func validateConnectorsConfig(mp map[string]ConnectorConfig) error {
+	var err, pErr error
+	for k, cfg := range mp {
+		if cfg.Plugin == "" {
```

Comment:
Do we want to check for blank strings too (i.e. only whitespace)?

Reply:
I tried that before by hand.. an empty field in the yaml file would be parsed as "", this is also the case when the field value is a couple of spaces like "    ", they both equal "" after parsing.

Reply:
Awesome, it's great that you checked that already!